daemon/exec_linux.go

```diff
@@ -41,6 +41,7 @@ func getUserFromContainerd(ctx context.Context, containerdCli *containerd.Client
 		coci.WithUser(ec.User),
 		withResetAdditionalGIDs(),
 		coci.WithAdditionalGIDs(ec.User),
+		coci.WithAppendAdditionalGroups(ec.Container.HostConfig.GroupAdd...),
```

Comment:
Somewhat orthogonal to this PR, but looking at the code, I went a bit down the rabbit-hole 🐰 ;

`execSetPlatformOpt` also has a Windows implementation, and it looks like we're currently fully ignoring these options on Windows; disclaimer: perhaps we document these to not be supported, and perhaps we don't support them at the API level; https://github.com/moby/moby/blob/a66ddb7c59781ab4b7d8b931658fa315157bc29b/daemon/exec_windows.go#L11-L16

_However_, I see this function is _also_ stubbed out in containerd; I'm wondering if we should call it in the Windows counterpart (and leave it up to containerd's implementation to do what's needed) ❓ ; it's a bit ambiguous though, because the comment describes LCOW (not "WCOW" :thinking:)
https://github.com/moby/moby/blob/a66ddb7c59781ab4b7d8b931658fa315157bc29b/vendor/github.com/containerd/containerd/oci/spec_opts.go#L812-L820

Further down the rabbit-hole; the `WithUser` and `WithUIDGID` functions look really scary, as they **unconditionally reset `AdditionalGids`** that means that _order of these options matters_. ⚠️ 
https://github.com/moby/moby/blob/a66ddb7c59781ab4b7d8b931658fa315157bc29b/vendor/github.com/containerd/containerd/oci/spec_opts.go#L594-L602
https://github.com/moby/moby/blob/a66ddb7c59781ab4b7d8b931658fa315157bc29b/vendor/github.com/containerd/containerd/oci/spec_opts.go#L697-L702

- On our side, we should consider adding a comment; it's currently somewhat clear because we _also_ call our own `withResetAdditionalGIDs` (which resets the field)
- ^^ but (looking at the code above), it looks like that function is redundant, so we can remove it (?)
- We should contribute a patch to containerd to _at least_ update the GoDoc for these functions to call that out
- _Perhaps_ consider making it an error-condition when setting a `User` and `AdditionalGids` is _not empty_ (?)

Reply:
Note: we had to reset the additional GIDs because containerd didn't do it at the time this was implemented, containerd changed its implementation since, was added here: https://github.com/moby/moby/commit/174802e15f12ee4384c7672676247aaf28997f76#diff-ebde8092f4da3b1f60a5d5b7d805276d2c831362b528c385e8792a9168a34444

So yeah, we can remove this call now I guess, maybe in a new PR

Reply:
Will do! :)

Reply:
> Perhaps consider making it an error-condition when setting a User and AdditionalGids is not empty (?)

In `WithUser`, or on our side?

Reply:
> In `WithUser`, or on our side?

In another PR :) And surely with an issue so that we can discuss about this in a more approriate place